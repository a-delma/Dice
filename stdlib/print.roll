import "../stdlib/math.roll"; 

[] -> Void newline;
[] -> Void dot;
[Int] -> Void printDigit;
[Int] -> Void printInt;
[Float] -> Float mantissa;
[Float] -> Void printMantissa;
[Float] -> Void printFloat;

//how many decimal points to print for floats
Int floatPrecision;
floatPrecision = 4;

newline = lambda()->Void{ putChar(10); };
dot = lambda()->Void{ putChar(46); };

//Only works on single digits, for internal use
//  to use with printInt
printDigit = lambda(Int d)->Void{
  putChar(48 + d);
};

printInt = lambda(Int n)->Void{
  if(n < 10){
    printDigit(n);
  } else {
    printDigit(mod(n, 10));
    self(n/10); 
  }
};

mantissa = lambda()->Float{
  //stub
  return 9.234;
}

//should take a float less than 1.
printMantissa = lambda(Float f, Int remaining)->Void{
  if(remaining < 1){
    //just return 
  } else {
    printDigit(floatToInt(f * 10));
    //self( , remaining - 1);
  }
};

printFloat = lambda(Float f)->Void{
  Int theInt;
  theInt = floatToInt(f);
  printInt(theInt);
  dot();
};
