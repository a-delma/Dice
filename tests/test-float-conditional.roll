import "../stdlib/print.roll";

Int a;
setSeed(1);
for (a = 1; a < 100; a = a + 1){
    if (0.333) 
    printInt(1);
    else 
    printInt(0);
}
//Printing stuff//////////////
newline();
newline();
putChar(78);
putChar(69);
putChar(87);
putChar(32);
putChar(83);
putChar(69);
putChar(69);
putChar(68);
newline();
newline();
//////////////////////////////
setSeed(2);
for (a = 1; a < 100; a = a + 1){
    if (0.333) 
    printInt(1);
    else 
    printInt(0);
}
