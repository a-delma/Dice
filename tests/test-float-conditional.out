011001010110100000000101000000000000001000011000000110101011000101010000001100100101110100000100000

NEW SEED

101000000000001100111100000100000010111110001101101000000010101110010000001000101000000110000110000